src/Chapter1.hs

```diff
@@ -553,9 +555,12 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
-
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | (x > y && x < z) || (x > z && x < y) = x
+    | (y > x && y < z) || (y > z && y < x) = y
+    | (z > x && z < y) || (z > y && z < x) = z
+    | otherwise = if x == y then x else z
```

Comment:
I guess if you use `<=` `>=` in the above checks, you won't need the last check here, and could get rid of the third clause as well (and return `z` straightaway) 👌🏼